packages/terra-functional-testing/src/terra-cli/wdio/test-runner.js

```diff
@@ -29,39 +31,69 @@ class TestRunner {
    * @param {string} options.config - A file path to the test runner configuration.
+   * @param {boolean} options.disableSeleniumService - A flag to disable the selenium docker service.
+   * @param {string} options.externalHost - The host address the testing environment is connected to.
+   * @param {number} options.externalPort - The port mapping from the host to the container.
    * @param {string} options.formFactor - A form factor for the test run.
-   * @param {string} options.locale - A language locale for the test run.
-   * @param {string} options.theme - A theme for the test run.
+   * @param {string} options.gridUrl - The remote selenium grid address.
    * @param {string} options.hostname - Automation driver host address.
+   * @param {boolean} options.keepAliveSeleniumDockerService - Determines to keep the selenium docker service running upon test completion.
+   * @param {string} options.locale - A language locale for the test run.
    * @param {number} options.port - Automation driver port.
-   * @param {string} options.baseUrl - The base url.
-   * @param {array} options.suite - Overrides specs and runs only the defined suites.
+   * @param {string} options.site - A file path to a static directory of assets. When defined, an express server will launch to serve the assets and disable running webpack..
    * @param {array} options.spec - A list of spec file paths.
-   * @param {Object} options.launcherOptions - Custom configuration launcher options. i.e. keepAliveSeleniumDockerService; updateScreenshots.
+   * @param {array} options.suite - Overrides specs and runs only the defined suites.
+   * @param {string} options.theme - A theme for the test run.
+   * @param {boolean} options.updateScreenshots - Updates all reference screenshots with the latest screenshots.
    * @returns {Promise} A promise that resolves with the test run exit code.
    */
   static async run(options) {
     let exitCode;
 
     try {
       const {
+        assetServerPort,
+        baseUrl,
+        browsers,
         config,
+        disableSeleniumService,
+        externalHost,
+        externalPort,
         formFactor,
+        gridUrl,
+        hostname,
+        keepAliveSeleniumDockerService,
         locale,
+        site,
         theme,
-        ...additionalLauncherOptions // hostname, port, baseUrl, suite, spec, launcherOptions
+        updateScreenshots,
+        ...cliOptions // spec, suite,
       } = options;
 
-      process.env.LOCALE = locale;
-      process.env.THEME = theme;
-
-      if (formFactor) {
-        process.env.FORM_FACTOR = formFactor;
-      }
+      const launcherOptions = {
+        ...cliOptions,
+        baseUrl: `http://${externalHost || getIpAddress()}:${externalPort || 8080}`,
+        capabilities: getCapabilities(browsers, !!gridUrl),
+        hostname: gridUrl || hostname || 'localhost',
+        port: gridUrl ? 80 : 4444,
+        launcherOptions: {
```

Comment:
Thinking about this a little more, I'm wondering if it would be better to define the services here. Services here would overwrite the default services in the wdio config and we could remove the custom property merges in the services constructors. We'd still probably need the launcher options for the onComplete hook to pass down test run configuration the reporter's merge function. 

Caveats (with either solution) is that Jenkins will need to continue to run the `wdio` command. It would not be able to use the test runner cli directly while it is still using envs. We might be able to make Jenkins changes will wdio v7 to remove all envs.

Reply:
I like this idea! I assume with this we will continue to keep how the services are defined as they are now in wdio.conf.

Reply:
That sounds right, I think with these changes we won't need to make many modifications to the current wdio.conf. The terra-cli will define any necessary configurations that will take precedence over the wdio.conf, but we can leave the current wdio.conf as a default for Jenkins to use and configure with ENVs. Then we can passively switch Jenkins over to the terra-cli and circle back on a major version bump and delete all ENVs from the default wdio.conf.

Reply:
+1 to consolidating the merges

Reply:
Added the services to the launcher: https://github.com/cerner/terra-toolkit/pull/583/commits/eb6e22e962d6bb12f197d735445237e6d2aaa5ff